### PR TITLE
Inventory and Item

### PR DESCRIPTION
**Pull request by siddharthgowda**
I accidentally created a new branch of Inventory and Item (so I deleted the old branch and its pull request)

Essentially here I am just making the entities for items (and its subclasses) and inventory. I also made a class that contains the name and description of item subclasses because those names will be needed in multiple classes (specifically the InventoryInteractor). I also made some minor refactoring changes (made a subfolder called items).

### CODE REVIEW

**Review by sbrad044 — DISMISSED** [review]

File: src/main/java/com/mg105/entities/Item.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
+package com.mg105.entities;
+
+
+public abstract class Item {
+    final public String NAME;
+    final public String DESCRIPTION;
```
Comment (6167656e74323431) [inline src/main/java/com/mg105/entities/Item.java:6]: `ALL_CAPS` is for public final static things (true constants).

**Review by 6167656e74323431 — CHANGES_REQUESTED** [review]

File: src/main/java/com/mg105/entities/Inventory.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        assert isInventoryFull() : new RuntimeException("Item cannot be added when Inventory is full");
+        this.items.add(item);
+    }
+
+    public boolean removeItem(String itemName) {
+        for (int i = 0; i < numberOfItems(); i++) {
+            if (this.items.get(i).getName().equals(itemName)) {
+                this.items.remove(i);
+                return true;
+            }
```
Comment (hoodiexxx) [inline src/main/java/com/mg105/entities/Inventory.java:34]: smart implementation

**Review by hoodiexxx — DISMISSED** [review]
good coding

**Review by 6167656e74323431 — APPROVED** [review]
YES!

**Review by 6167656e74323431 — APPROVED** [review]

**Review by hoodiexxx — DISMISSED** [review]
The code looks well made and it seems cover all the corner cases in the InventoryTest.java.

File: src/main/java/com/mg105/entities/items/HealthPotion.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
+package com.mg105.entities.items;
+
+import com.mg105.entities.Item;
+import com.mg105.utils.ItemConstants;
+
+public class HealthPotion extends Item {
+
+    // This number could change later
+
+    public final static int HEALING_POINTS = 25;
```
Comment (zhaoan12) [inline src/main/java/com/mg105/entities/items/HealthPotion.java:10]: Nice class! After playtesting our game we can always change the value of healing points to provide a balanced game for the player.

**Review by zhaoan12 — DISMISSED** [review]
Well made classes.

**Review by m-grigoriu — DISMISSED** [review]
Everything looks good. There should eventually be a useItem() method, either in the Item class or the Inventory class (depending on which implementation is preferable). I am also not sure how useful a separate file for item constants is, unless we plan on having a large number of items and want one file where we can see every detail at once.

**Review by 6167656e74323431 — APPROVED** [review]
Fine.

Although please create Github issues for missing documentation/tests.

**Review by 6167656e74323431 — APPROVED** [review]

**Review by zhaoan12 — APPROVED** [review]

**Review by sbrad044 — APPROVED** [review]

**Comment by 6167656e74323431** [comment]
Merged during weekly meeting.

**Pull request merged**